tests/mpls/test_mpls.py

```diff
@@ -0,0 +1,330 @@
+        testutils.send(ptfadapter, src_pid, pkt)
+        try:
+            res=testutils.verify_packet_any_port(ptfadapter, exp_pkt, ports=[dst_pid])
+            logger.info(res)
+        except Exception as e:
+            self.teardown_labels(setup)  
+            pytest.fail('MPLS pop test failed \n'+ str(e))
+        
+        self.teardown_labels(setup)
+
+    def test_swap_label(self, setup, ptfadapter):
+        """ test swap label """
+
+        dst_pid=setup['dst_pid']
+        src_pid=setup['src_pid']
+
+        self.config_interface_mpls(setup, LABEL_SWAP_ROUTES) 
+   
+        time.sleep(2)
+ 
+        pkt=self.mpls_packet(setup, ptfadapter)
+        exp_pkt=self.expected_mask_mpls_swap_packet(pkt, 1000002)
+        
+        ptfadapter.dataplane.flush()
+        testutils.send(ptfadapter, src_pid, pkt)
+        try:
+            res=testutils.verify_packet_any_port(ptfadapter, exp_pkt, ports=[dst_pid])
+            logger.info(res)
+        except Exception as e:
+            self.teardown_labels(setup)
+            pytest.fail('MPLS swap test failed \n' + str(e)) 
+      
+        self.teardown_labels(setup)
+ 
+    def test_push_label(self, setup, ptfadapter):
+        """ test push label """
+        
+        dst_pid=setup['dst_pid']
+        src_pid=setup['src_pid']
+
+        self.config_interface_mpls(setup, LABEL_PUSH_ROUTES)
+
+        time.sleep(2)
+
+        pkt=self.icmp_packet(setup, ptfadapter)
+        epkt=pkt.copy()
+        pkt1=epkt['IP']
+        epkt['Ethernet'].type=0x8847
+        epkt['Ethernet'].remove_payload()
+        mp=MPLS(label=1000002, s=1, ttl=255)
+        mp.remove_payload()
+        epkt /= mp
+        epkt /= pkt1
+        exp_pkt=self.expected_mask_mpls_push_packet(epkt, 1000001)
+
+        ptfadapter.dataplane.flush()
+        testutils.send(ptfadapter, src_pid, pkt)
+
+        try:
+            res=testutils.verify_packet_any_port(ptfadapter, exp_pkt, ports=[dst_pid])
+            logger.info(res)
+        except Exception as e:
+            self.teardown_labels(setup)
+            pytest.fail('MPLS push test failed \n' + str(e))
+
+        self.teardown_labels(setup)
+
+    def test_swap_labelstack(self, setup, ptfadapter):
+        """ test swap labelstack """
+
+        dst_pid=setup['dst_pid']
+        src_pid=setup['src_pid']
+
+        self.config_interface_mpls(setup, LABEL_SWAP_ROUTES)
+ 
+        time.sleep(2)
+
+        pkt=self.mpls_stack_packet(setup, ptfadapter)
+        exp_pkt=self.expected_mask_mpls_swap_packet(pkt, 1000002)
+
+        ptfadapter.dataplane.flush()
+        testutils.send(ptfadapter, src_pid, pkt)
+
+        try:
+            res=testutils.verify_packet_any_port(ptfadapter, exp_pkt, ports=[dst_pid])
+            logger.info(res)
+        except Exception as e:
+            self.teardown_labels(setup)
+            pytest.fail('MPLS swap labelstack test failed \n' + str(e))
+
+        self.teardown_labels(setup)
+
+class TestBasicMpls(BaseMplsTest):
```

Comment:
It seems that this subclass does nothing extra. Why not use single plain class?

Reply:
Can you combine `BaseMplsTest` and `TestBasicMpls` into a single plain class?

Reply:
The error should have been caused by `__metaclass__=ABCMeta` and definitions of unused methods like `def setup_rules` and `def post_setup_hook`. You can refer to other plain test class, for example: `sonic-mgmt/tests/arp/test_neighbor_mac.py`.

Detailed steps:
1. Remove class `TestBasicMpls`.
2. Rename `BaseMplsTest` to `TestBasicMpls`
3. Remove `__metaclass__=ABCMeta`
4. Remove methods `def setup_rules` and `def post_setup_hook`.

The reason of this ask is that using meta class in this script is an overdesign. A simple plain test class is enough. There is no need for meta class.

Reply:
Changes done.